pylint/config/argument.py

```diff
@@ -8,25 +8,43 @@
 """
 
 
+import argparse
 import re
-from typing import Callable, Dict, List, Optional, Pattern, Union
+from typing import Callable, Dict, List, Optional, Pattern, Sequence, Union
 
 from pylint import utils as pylint_utils
 
-_ArgumentTypes = Union[str, List[str], int, Pattern[str]]
+_ArgumentTypes = Union[str, Sequence[str], int, Pattern[str], bool]
 """List of possible argument types."""
 
 
-def _csv_transformer(value: str) -> List[str]:
+def _csv_transformer(value: str) -> Sequence[str]:
     """Transforms a comma separated string."""
     return pylint_utils._check_csv(value)
 
 
+YES_VALUES = {"y", "yes", "true"}
+NO_VALUES = {"n", "no", "false"}
+
+
+def _yn_transformer(value: str) -> bool:
+    """Transforms a yes/no or stringified bool into a bool."""
+    value = value.lower()
+    if value in YES_VALUES:
+        return True
+    if value in NO_VALUES:
+        return False
+    raise argparse.ArgumentError(
```

Comment:
Will be covered as soon as we flip the switch. `optparse` now raises a warning here.